python/pyarrow/tests/test_pandas.py

```diff
@@ -3325,6 +3325,21 @@ def test_cast_timestamp_unit():
     assert result.equals(expected)
 
 
+def test_nested_with_timestamp_tz_round_trip():
+    ts = pd.Timestamp.now()
+    ts_dt = ts.to_pydatetime()
+    arr = pa.array([ts_dt], type=pa.timestamp('us', tz='America/New_York'))
```

Comment:
Does this test presume that `ts` itself was produced in "America/New York" timezone? It's not clear to me.

Reply:
I don't believe so. Pretty sure my machine uses it. Local time, I'll double check by setting a different tz